ts/features/payments/checkout/screens/WalletPaymentOutcomeScreen.tsx

```diff
@@ -39,18 +42,41 @@ type WalletPaymentOutcomeRouteProps = RouteProp<
 const WalletPaymentOutcomeScreen = () => {
   useAvoidHardwareBackButton();
 
+  const dispatch = useIODispatch();
   const { params } = useRoute<WalletPaymentOutcomeRouteProps>();
   const { outcome } = params;
 
   const navigation = useIONavigation();
   const paymentDetailsPot = useIOSelector(walletPaymentDetailsSelector);
   const onSuccessAction = useIOSelector(walletPaymentOnSuccessActionSelector);
   const profileEmailOption = useIOSelector(profileEmailSelector);
+  const paymentOngoingHistory = useIOSelector(selectOngoingPaymentHistory);
 
   const supportModal = usePaymentFailureSupportModal({
     outcome
   });
 
+  useOnFirstRender(() => {
```

Comment:
What do you think of simplifying this function to 
```
const kind = outcome === SUCCESS ? "COMPLETED" : outcome === DUPLICATE_ORDER ? "DUPLICATED" : undefined
const rptId = paymentOngoingHistory?.rptId

if (!kind || !rptId) {
  return
}

dispatch(
  paymentCompletedSuccess({
    rptId,
    kind
  })
);
```

Reply:
I like it, it's more concise code!

Addressed into 6949c03f806b3ac6b224a05e95987e4b825887f4